File: infix_to_postfix5.c

#include <stdio.h>
#include <ctype.h>
#define MAX 100
char stack[MAX];int top=-1;
void push(char c){stack[++top]=c;}
char pop(){return stack[top--];}
int prec(char c){if(c=='+'||c=='-')return 1;if(c=='*'||c=='/')return 2;return 0;}
int main(){
    char infix[MAX],postfix[MAX];printf("Enter infix: ");scanf("%s",infix);
    int k=0;
    for(int i=0;infix[i];i++){
        char c=infix[i];
        if(isalnum(c))postfix[k++]=c;
        else if(c=='(')push(c);
        else if(c==')'){while(top!=-1&&stack[top]!='(')postfix[k++]=pop();top--;}
        else{while(top!=-1&&prec(stack[top])>=prec(c))postfix[k++]=pop();push(c);}
    }
    while(top!=-1)postfix[k++]=pop();
    postfix[k]='\0';printf("Postfix: %s\n",postfix);return 0;
}

File: Infix To Postfix 5.py
def infix_to_postfix(expression):
    precedence = {'^': 3, '*': 2, '/': 2, '%': 2, '+': 1, '-': 1}
    associativity = {'^': 'right', '*': 'left', '/': 'left', '%': 'left', '+': 'left', '-': 'left'}
    stack = []
    output = []

    tokens = list(expression.replace(" ", ""))

    for token in tokens:
        if token.isalnum():  # Operand (letters or digits)
            output.append(token)
        elif token == '(':
            stack.append(token)
        elif token == ')':
            # Pop until matching '('
            while stack and stack[-1] != '(':
                output.append(stack.pop())
            stack.pop()  # Remove '('
        else:
            # Operator
            while (stack and stack[-1] != '(' and
                   (precedence[stack[-1]] > precedence[token] or
                    (precedence[stack[-1]] == precedence[token] and associativity[token] == 'left'))):
                output.append(stack.pop())
            stack.append(token)

    # Pop remaining operators
    while stack:
        output.append(stack.pop())

    return ''.join(output)
expressions = [
    "A+B",
    "A+B*C",
    "(A+B)*C",
    "A^B^C",
    "A*(B+C)/D",
    "A+B*C-D/E"
]

for expr in expressions:
    print(f"Infix: {expr} → Postfix: {infix_to_postfix(expr)}")
